server/controllers/threads.controller.js

```diff
@@ -0,0 +1,179 @@
+import httpStatus from 'http-status';
+import Promise from 'bluebird';
+import db from '../../config/sequelize';
+import APIError from '../helpers/APIError';
+
+const Message = db.Message;
+const Thread = db.Thread;
+const User = db.User;
+
+/**
+ * Start a new thread
+ * @property {Array} req.body.participants - Array of usernames to include in the thread.
+ * @property {string} req.body.subject - Subject line of the message
+ * @property {string} req.body.message - Body of the message
+ * @property {string} req.body.topic - Topic of the thread
+ * @returns {Message}
+ */
+function create(req, res, next) {
+    const { username } = req.user;
+    const participants = req.body.participants;
+    let users = [];
+    let userPromises = [];
+    let date = new Date();
+    participants.forEach((participant) => {
+      let userPromise = User.findOrCreate({where: {username: participant}})
+      .spread((user, created) => {
+        users.push(user);
+      });
+      userPromises.push(userPromise);
+    });
+
+    Promise.all(userPromises).then(() => {
```

Comment:
Promise.all resolves with an array of the results, so here and above can be shortened to `Promise.all(participants.map(participant => User.findOrCreate({where: {username: participant}}))).then(users => {`